src/librustc_mir/transform/generator.rs

```diff
@@ -110,7 +128,13 @@ impl<'tcx> MutVisitor<'tcx> for DerefArgVisitor {
                 projection: Box::new([ProjectionElem::Deref]),
             });
         } else {
-            self.super_place(place, context, location);
+            self.visit_place_base(&mut place.base, context, location);
+
+            for elem in place.projection.iter() {
+                if let PlaceElem::Index(local) = elem {
+                    assert_ne!(*local, self_arg());
```

Comment:
```suggestion
                    self.visit_local(local);
```

Reply:
Can't do this because `visit_local` takes a `&mut` and the idea is that we will be passing immutable things around due to interning.